src/parsers/texttracks/webvtt/parse_mp4_embedded_wvtt.ts

```diff
@@ -0,0 +1,178 @@
+      } else if (currentBoxName === "vtte") {
+        if (payloadSize > 8) {
+          mdatOffset += payloadSize - 8;
+        }
+      } else {
+        log.error("webvtt: encountered unknown fragmented vtt box: ", currentBoxName);
+        mdatOffset += Math.min(payloadSize - 8, 1);
+      }
+
+      if (duration > 0) {
+        if (currentBoxData !== null) {
+          const cue = parseVttC(
+            currentBoxData,
+            timeOffset + startTime / timescale,
+            timeOffset + lastTime / timescale,
+            cueProducer,
+          );
+          if (cue !== null) {
+            cuesArray.push(cue);
+          }
+        }
+      } else {
+        log.error("webvtt: cue duration missing");
+      }
+    }
+  }
+
+  if (mdatOffset !== mdat.length) {
+    log.error("WEBVTT: end offset is not equal to mdat length", mdatOffset, mdat.length);
+  }
+
+  return cuesArray;
+}
+
+export type ICueProducerFunction<T> = (
+  cueObj: {
+    start: number;
+    end: number;
+    settings: Partial<Record<string, string>>;
+    header?: string | undefined;
+    payload: string[];
+  },
+  styling: { classes: IStyleElements; global?: string | undefined },
+) => T | null;
+
+/**
+ * Parse VTT cue data as found in an "vttc" mp4 box.
+ * @param {Uint8Array} data
+ * @param {number} startTime
+ * @param {number} endTime
+ * @returns {Object|null}
+ */
+function parseVttC<T>(
+  data: Uint8Array,
+  startTime: number,
+  endTime: number,
+  cueProducer: ICueProducerFunction<T>,
+): T | null {
+  const payload = getPayl(data);
+  const settings = getSttg(data);
+  // const iden = getIden(data);
+  if (payload === null) {
+    return null;
+  }
+  const cue = cueProducer(
+    {
+      start: startTime,
+      end: endTime,
+      settings: settings !== null ? parseSettings(utf8ToStr(settings)) : {},
+      header: undefined,
+      payload: [utf8ToStr(payload)],
+    },
+    {
+      classes: {},
+      global: undefined,
+    },
+  );
+  return cue;
+}
+
+/**
+ * Returns the content of the first "payl" box encountered in the given ISOBMFF
+ * data.
+ * Returns null if not found.
+ * @param {Uint8Array} buf
+ * @returns {Uint8Array|null}
+ */
+function getPayl(buf: Uint8Array): Uint8Array | null {
+  return getBoxContent(buf, 0x7061796c /* "payl" */);
+}
+/**
+ * Returns the content of the first "sttg" box encountered in the given ISOBMFF
+ * data.
+ * Returns null if not found.
+ * @param {Uint8Array} buf
+ * @returns {Uint8Array|null}
+ */
+function getSttg(buf: Uint8Array): Uint8Array | null {
+  return getBoxContent(buf, 0x73747467 /* "sttg" */);
+}
+
+// TODO: how to exploit that one in the current API?
+// /**
+//  * Returns the content of the first "iden" box encountered in the given ISOBMFF
+//  * data.
+//  * Returns null if not found.
+//  * @param {Uint8Array} buf
+//  * @returns {Uint8Array|null}
+//  */
+// function getIden(buf: Uint8Array): Uint8Array | null {
+//   return getBoxContent(buf, 0x6964656e /* "iden" */);
+// }
```

Comment:
is it needed ?

Reply:
I think it's for an identifier that may be of use to an application.

For now we have no way to translate it through the API but I left that commented code basically as a long term TODO to show that this is a possible future API whose implementation has already been done and thought about.

But I understand that littering with commented code seems not optimal, I'll do a supplementary commit removing that code.

Reply:
done